plugins/thunderbolt-power/fu-plugin-thunderbolt-power.c

```diff
@@ -240,11 +240,23 @@ fu_plugin_update_prepare (FuPlugin *plugin,
 		return FALSE;
 
 	data->needs_forcepower = TRUE;
+
 	/* wait for the device to come back onto the bus */
 	fu_device_set_status (device, FWUPD_STATUS_DEVICE_RESTART);
-	g_usleep (TBT_NEW_DEVICE_TIMEOUT * G_USEC_PER_SEC);
+	for (guint i = 0; i < 10; i++) {
```

Comment:
So this is effectively waiting 4 minutes (25s * 10 tries)?  That seems pretty long to me.  If it doesn't come back in the first minute it's probably not going to I would say.

Reply:
10x250000 = 10x250ms = 2.5s